### ISSUE TITLE
Add a function to profile likelihood

### ISSUE THREAD

**Issue by schuemie** [OHDSI/Cyclops#48]
Would it be possible to add a function that returns the (log) likelihood for a vector of coefficient values? Ideally, the function would work similar to the [confint function](http://ohdsi.github.io/Cyclops/reference/confint.cyclopsFit.html), so having the following arguments:

- **object**: A fitted Cyclops model object
- **parm**:  A specification of which parameter requires profiling; either a number or a covariateId name
- **x**: vector of values of the parameter for which we want to compute the likelihood
- **includePenalty**: Logical: Include regularized covariate penalty in profile

A special case might be an unfitted model with only 1 independent variable. In that case it would be nice if it wasn't necessary to first fit the model (since an optimum may not exist).

RIght now I have to use [this workaround](https://github.com/ohdsi-studies/SmallSampleEvidenceSynthesis/blob/master/R/RealWorldExamples.R#L273), which works but is very slow.

**Comment by msuchard** [comment]
Am now working on this in 8e703f6f428e4b33b126e1b5b57f04eb2cef3e9d .  Sorry for the delay!

**Comment by msuchard** [comment]
@schuemie , could you check out https://github.com/OHDSI/Cyclops/blob/7103e9e323283235dfe34c6feb2515c041acf5cf/tests/testthat/test-profileLikelihood.R#L41 as an example of evaluating the profile likelihood along a grid values?   The highlighted line illustrates how to construct a 1D `cyclopsModelFit` without having to find the (possibly nonexistent) optimum first.

Also, if you specify a number of threads into the `cyclopsControl` block, then the code should evaluate the grid points in parallel.

Once you are happy with the functionality, please merge into `master`.

**Comment by schuemie** [comment]
Thanks! I see a seven-fold increase in speed in my application ;-)

Merged into master. Closing now.
